primo-zip.py

```diff
@@ -0,0 +1,97 @@
+# importing required modules
+import shutil
+import fileinput
+import json
+import os
+
+
+def set_secrets_and_zip(secret_data, types_to_check, base_directory, directory, zip_name):
+
+    # var for file paths list
+    file_paths = []
+
+    # crawl the directory structure
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            if (filename.endswith(types_to_check)):
+                filepath = os.path.join(root, filename)
+                file_paths.append(filepath)
+
+                # find and replace each secret tag with the actual secret
+                cnt = 0
+                for i in secret_data['secret_details']:
+                    secret_name = str(
+                        secret_data['secret_details'][cnt]['secret_name'])
+                    secret_value = str(
+                        secret_data['secret_details'][cnt]['secret_value'])
+
+                    with fileinput.FileInput(filepath, inplace=True) as file:
+                        for line in file:
+                            print(line.replace(
+                                str(secret_name), secret_value), end='')
+                    cnt += 1
+
+    # zip all files after inserting the secret(s)
+    shutil.make_archive(zip_name, 'zip', base_directory, directory)
+
+
+def clear_secrets(secret_data, types_to_check, directory):
+
+    # var for file paths list
+    file_paths = []
+
+    # crawl the directory structure
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            if (filename.endswith(types_to_check)):
+                filepath = os.path.join(root, filename)
+                file_paths.append(filepath)
+
+                # find and replace each actual secret with the secret tag
+                cnt = 0
+                for i in secret_data['secret_details']:
+                    secret_name = str(
+                        secret_data['secret_details'][cnt]['secret_name'])
+                    secret_value = str(
+                        secret_data['secret_details'][cnt]['secret_value'])
+
+                    with fileinput.FileInput(filepath, inplace=True) as file:
+                        for line in file:
+                            print(line.replace(
+                                secret_value, str(secret_name)), end='')
+                    cnt += 1
+
+
+def main():
+    # zip info
+    types_to_check = ('.js', '.css', 'html', 'md', 'txt')
+    base_directory = '.'
+    directory = './01UCS_LAL-UCLA'
```

Comment:
@darrowcoucla Instead of multiple numbered (and unnumbered) variables, I suggest replacing `directory` and `directory2` with `directory_prod` & `directory_test` (if you prefer the alphabetically similar names), or `prod_directory` & `test_directory` if you prefer that style.  Same with the `zip_name` variables.  That makes the intent of tests like the one at lines 74-76 clearer.

Then lines 77-78 can be
```
directory = test_directory
zip_name = test_zip_name
# or however you renamed those variable
```

Or simpler, keep `directory` and `zip_name` as defined on lines 69 and 71, and assign the test values in the conditional at lines 76-78.  No need for the `prod` variable names I initially suggested, given there are just the 2 cases.

Reply:
Names updated.